samples/seh/main.c

```diff
@@ -0,0 +1,44 @@
+#include <hal/debug.h>
+#include <hal/video.h>
+#include <windows.h>
+
+// The nullpointer deref is in its own function, because LLVM can't catch
+// asynchronous exceptions in the same frame that they're thrown in. This is a
+// documented MSVC-incompatibility in LLVM, see:
+// https://clang.llvm.org/docs/MSVCCompatibility.html#abi-features
+static void nullptr_deref (void)
+{
+  int *p = NULL;
+  *p = 5;
+}
+
+int main(void)
+{
+    XVideoSetMode(640, 480, 32, REFRESH_DEFAULT);
+
+    __try {
+        __try {
+            __try {
```

Comment:
I think you should add some vertical spaces between these, or move some things into functions.

All this nesting with the same keywords isn't very readable in my opinion.